app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,49 @@
+
+class Api::V1::BaseController < ApplicationController
+  before_action :retrieve_source_model_by_id, only: [:edit, :update, :destroy, :show]
+
+  def index
+    @source_models = source_klass.all
```

Comment:
shouldn't the show/index/create/update functions here be the ones to render json?